src/main/java/com/microsoft/sqlserver/jdbc/SQLServerMSAL4JUtils.java

```diff
@@ -84,10 +89,17 @@ static SqlAuthenticationToken getSqlFedAuthToken(SqlFedAuthInfo fedAuthInfo, Str
         lock.lock();
 
         try {
+            String hashedSecret = getHashedSecret(new String[] {fedAuthInfo.stsurl, user, password});
+            PersistentTokenCacheAccessAspect persistentTokenCacheAccessAspect;
+
+            if (null == (persistentTokenCacheAccessAspect = tokenCacheMap.getEntry(hashedSecret))) {
+                persistentTokenCacheAccessAspect = new PersistentTokenCacheAccessAspect();
+                tokenCacheMap.addEntry(hashedSecret, persistentTokenCacheAccessAspect);
+            }
```

Comment:
Having assignment and comparison together is a bit hard to parse. I'd prefer this.
```suggestion
            PersistentTokenCacheAccessAspect persistentTokenCacheAccessAspect = tokenCacheMap.getEntry(hashedSecret);

            if (null == persistentTokenCacheAccessAspect) {
                persistentTokenCacheAccessAspect = new PersistentTokenCacheAccessAspect();
                tokenCacheMap.addEntry(hashedSecret, persistentTokenCacheAccessAspect);
            }
```